Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -238,6 +294,8 @@ static u32 CalculateLevelSize(u32 level_0_size, u32 level)
 // Used by TextureCache::Load
 TextureCache::TCacheEntryBase* TextureCache::ReturnEntry(unsigned int stage, TCacheEntryBase* entry)
 {
+	entry->do_partial_texture_updates();
```

Comment:
Ok, i wrote a small test program for this, and this is the debug output:
[code]
11:05:590 D3DBase.cpp:113 N[Video]: Successfully loaded D3DCompiler_42.dll. If you're having trouble, try updating your DX runtime first.
11:05:904 BPStructs.cpp:676 W[Video]: Unknown BP opcode: address = 0x00000069 value = 0x000004ed
11:05:904 BPStructs.cpp:676 W[Video]: Unknown BP opcode: address = 0x00000046 value = 0x00000273
11:49:874 TextureCache.cpp:183 E[Video]: efb copy offset: 0x10334860
11:49:874 TextureCache.cpp:183 E[Video]: efb copy offset: 0x10344760
11:49:875 TextureCache.cpp:183 E[Video]: efb copy offset: 0x10334760
11:49:875 TextureCache.cpp:183 E[Video]: efb copy offset: 0x10343860
11:49:875 TextureCache.cpp:183 E[Video]: efb copy offset: 0x10343960
11:49:875 TextureCache.cpp:183 E[Video]: efb copy offset: 0x10343a60
11:49:889 TextureCacheBase.cpp:418 E[Video]: normal texture offset: 0x10332960
11:49:895 TextureCacheBase.cpp:259 E[Video]: Doing partial texture update for a texture at: 0x10332960, efb copy at: 0x10334760, w/h: 1024/1024, update at: x/y:  960/   0, w/h:   32/  32
11:49:895 TextureCacheBase.cpp:259 E[Video]: Doing partial texture update for a texture at: 0x10332960, efb copy at: 0x10334860, w/h: 1024/1024, update at: x/y:  992/   0, w/h:   32/  32
11:49:895 TextureCacheBase.cpp:259 E[Video]: Doing partial texture update for a texture at: 0x10332960, efb copy at: 0x10343860, w/h: 1024/1024, update at: x/y:  480/  32, w/h:   32/  32
11:49:895 TextureCacheBase.cpp:259 E[Video]: Doing partial texture update for a texture at: 0x10332960, efb copy at: 0x10343960, w/h: 1024/1024, update at: x/y:  512/  32, w/h:   32/  32
11:49:895 TextureCacheBase.cpp:259 E[Video]: Doing partial texture update for a texture at: 0x10332960, efb copy at: 0x10343a60, w/h: 1024/1024, update at: x/y:  544/  32, w/h:   32/  32
11:49:895 TextureCacheBase.cpp:259 E[Video]: Doing partial texture update for a texture at: 0x10332960, efb copy at: 0x10344760, w/h: 1024/1024, update at: x/y:  960/  32, w/h:   32/  32
[/code]

So the efb copies are apparently done 1st, and then we load the texture in:
TextureCache::TCacheEntryBase* TextureCache::Load(const u32 stage)

So the way the pr currently is implemented seems fine to me. The textrue updates are uploaded by the game, then we load the texture, maybe rebuild it for a new world/stage, and then we apply the updates on top of it.

But ReturnEntry might really be a bad place for this. Should i move it to the places before calling ReturnEntry where it makes sense to try this?